src/schemas.jl

```diff
@@ -254,35 +254,57 @@ accepted_field_type(::SchemaVersion, ::Type{Symbol}) = Union{Symbol,String}
 """
     Legolas.find_violation(ts::Tables.Schema, sv::Legolas.SchemaVersion)
 
-For required field `f::F` of `sv`:
+Return first schema violation; otherwise, return `nothing`. For description of
+schema violations, see [`Legolas.find_violations`](@ref).
 
-- Define `A = Legolas.accepted_field_type(sv, F)`
-- If `f::T` is present in `ts`, ensure that `T <: A` or else immediately return `f::Symbol => T::DataType`.
-- If `f` isn't present in `ts`, ensure that `Missing <: A` or else immediately return `f::Symbol => missing::Missing`.
-
-Otherwise, return `nothing`.
-
-See also: [`Legolas.validate`](@ref), [`Legolas.complies_with`](@ref)
+See also: [`Legolas.find_violations`](@ref), [`Legolas.validate`](@ref), [`Legolas.complies_with`](@ref)
 """
 find_violation(::Tables.Schema, sv::SchemaVersion) = throw(UnknownSchemaVersionError(sv))
```

Comment:
I think we should deprecate this function in favor of `find_violations`

Reply:
done!